pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```
@@ -334,6 +349,133 @@ public RepositoryData generate() {
+                    new org.jboss.da.model.rest.GAV(
+                            bomConstraintGroupId,
+                            bomConstraintArtifactId,
+                            generationData.getBomVersion()),
+                    tempBuild);
+
+            final Artifact bom = new DefaultArtifact(
+                    bomConstraintGroupId,
+                    bomConstraintArtifactId,
+                    null,
+                    "pom",
+                    bomConstraintVersion);
+            final List<Dependency> bomConstraints = mvnResolver.resolveDescriptor(bom).getManagedDependencies();
+            if (bomConstraints.isEmpty()) {
+                throw new IllegalStateException("Failed to resolve " + bom);
+            }
+            // Must point to a text file which contains list of format "groupId:artifactId:version"
+            String extensionsListUrl = params.get("extensionsListUrl");
+
+            List<Artifact> extensionRtArtifactList = parseExtensionsArtifactList(extensionsListUrl);
+            for (Artifact extensionRtArtifact : extensionRtArtifactList) {
+                // this will resolve all the runtime dependencies and as a consequence populate the local Maven repo
+                // specified in the user settings.xml
+                DependencyResult result = mvnResolver.resolveManagedDependencies(
+                        extensionRtArtifact, // runtime extension artifact
+                        Collections.emptyList(), // enforced direct dependencies, ignore this
+                        bomConstraints, // version constraints from the BOM
+                        Collections.emptyList(), // extra maven repos, ignore this
+                        "test",
+                        "provided" // dependency scopes that should be ignored
+                );
+
+            }
+            return repackage(sourceDir);
+        } catch (Exception bme) {
+            bme.printStackTrace();
+            return null;
```

why is the exception swallowed here? Is there any case that bacon could continue after it?

---

`repositoryGeneration functions` dont throw the exceptions. Exceptions are handled/thrown by the calling function. None of the repository Generation method throws exception

---

that's not true, they throw `RuntimeException`s

---

take a look e.g. at `RepoBuilder`

---

Throwing Runtime exception now